core/SConscript.kernel

```diff
@@ -24,7 +31,7 @@ FEATURE_FLAGS = {
 }
 
 FEATURES_WANTED = ["input", "sd_card", "rgb_led", "dma2d", "consumption_mask", "usb" ,"optiga", "haptic"]
-if DISABLE_OPTIGA and PYOPT == '0':
+if DISABLE_OPTIGA and not PRODUCTION:
     FEATURES_WANTED.remove("optiga")
```

Comment:
1. The corresponding change was not made in SConscript.firmware. BTW, it sucks that there is so much code duplication between SConscript.firmware and SConscript.kernel. There must be a clean way to get rid of that.
2. Similar to line 20 where we handle `if STORAGE_INSECURE_TESTING_MODE and PRODUCTION`, wouldn't it be more appropriate to raise an error `if DISABLE_OPTIGA and PRODUCTION`, rather than silently ignore `DISABLE_OPTIGA` in production builds?
```suggestion
if DISABLE_OPTIGA:
    if PRODUCTION:
        raise RuntimeError("DISABLE_OPTIGA cannot be used in production")
    FEATURES_WANTED.remove("optiga")
```

Reply:
> 1. The corresponding change was not made in SConscript.firmware

that's because scons.firmware gets the `PYOPT` as a debug indicator, but it's fixed to 1 in kernel because of #4253. otherwise firmware's spelling is preferred -- not just "not production" but "yes debug" should be the condition for disabling optiga.

i can convert both sides to the runtime-error version though

Reply:
7e9a1f458766